ts/indicators/price-envelopes.src.ts

```diff
@@ -0,0 +1,339 @@
+/* *
+ *
+ *  License: www.highcharts.com/license
+ *
+ *  !!!!!!! SOURCE GETS TRANSPILED BY TYPESCRIPT. EDIT TS FILE ONLY. !!!!!!!
+ *
+ * */
+
+'use strict';
+
+import H from '../parts/Globals.js';
+
+/**
+ * Internal types
+ * @private
+ */
+declare global {
+    namespace Highcharts {
+
+        class PriceEnvelopesIndicator extends SMAIndicator {
+            public data: Array<PriceEnvelopesIndicatorPoint>;
+            public drawGraph(): void;
+            public init(): void;
+            public getValues(
+                series: Series,
+                params: PriceEnvelopesIndicatorOptionsParams
+            ): (boolean|IndicatorMultipleValuesObject);
+            public nameBase: string;
+            public options: PriceEnvelopesIndicatorOptions;
+            public parallelArrays: Array<string>;
+            public pointArrayMap: Array<string>;
+            public pointClass: typeof PriceEnvelopesIndicatorPoint;
+            public points: Array<PriceEnvelopesIndicatorPoint>;
+            public pointValKey: string;
+            public toYData(
+                point: PriceEnvelopesIndicatorPoint
+            ): [number, number, number];
+            public translate(): void;
+        }
+
+        interface PriceEnvelopesIndicatorGappedExtensionObject {
+            [index: string]: Dictionary<(number|undefined)>;

```

Comment:
If you go that far, you can remove the interface and use `Dictionary<Dictionary<(number|undefined)>>`.

Reply:
This same structure seems to appear in a few different indicators. Could it be better to define this as separate interfaces in one of the the indicators and the reuse the same in the other indicators, or do you think that they should be declared a separate interfaces per indicator?

Reply:
That is hard to tell, but as there is no mixin in use, I would go with separate interfaces.

Reply:
Added separate interfaces. I hope the naming is ok `PriceEnvelopesIndicatorGappedExtensionObject` and `PriceEnvelopesIndicatorGappedExtensionOptions`